controllers/operator/routes.go

```diff
@@ -328,17 +330,26 @@ func (r *AuthenticationReconciler) reconcileRoute(ctx context.Context, instance
 		if !IsRouteEqual(calculatedRoute, observedRoute) {
 			reqLogger.Info("Updating route")
 
-			observedRoute.ObjectMeta.Name = calculatedRoute.ObjectMeta.Name
-			observedRoute.ObjectMeta.Annotations = calculatedRoute.ObjectMeta.Annotations
+			observedRoute.Name = calculatedRoute.Name
+			observedRoute.Annotations = calculatedRoute.Annotations
 			observedRoute.Spec = calculatedRoute.Spec
 
-			err = r.Client.Update(ctx, observedRoute)
+			err = r.Update(ctx, observedRoute)
 			if err != nil {
 				reqLogger.Error(err, "Failed to update route")
 				return
 			}
 			*needToRequeue = true
 		}
+	} else {
+		reqLogger.Info("zenFrontDoor is enabled; deleting route")
+		err = r.Delete(ctx, observedRoute)
```

Comment:
This is an issue - if the Route doesn't exist, observedRoute is empty, meaning we hit a client error trying to delete something without naming it first.